sdk/core/System.ClientModel/src/Convenience/AsyncEnumerableResultOfT.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel;
+
+#pragma warning disable CS1591 // public XML comments
+public abstract class AsyncEnumerableResult<T> : ClientResult, IAsyncEnumerable<T>, IAsyncDisposable
+    where T : IPersistableModel<T>
+{
+    protected internal AsyncEnumerableResult(PipelineResponse response) : base(response)
+    {
+    }
+
+    public abstract IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default);
+
+    protected abstract ValueTask DisposeAsyncCore();
+
+    protected abstract void Dispose(bool disposing);
+
+    public async ValueTask DisposeAsync()
+    {
+        await DisposeAsyncCore().ConfigureAwait(false);
+
+        // Dispose of unmanaged resources. Note that DisposeAsyncCore disposes
+        // of managed resources asychronously -- we pass false to Dispose so we
+        // don't attempt to dispose of them synchronously as well.
+        Dispose(disposing: false);
```

Comment:
Seems like this should be an implementation detail of `DisposeAsyncCore` - I would expect that to have the same responsibility as calling `Dispose(true)`, which should do both managed and unmanaged resources.

My understanding is that `disposing: false` is intended to mean "we're not disposing, this is a destructor calling so just do the unmanaged junk" and isn't a path that would be called otherwise.

Reply:
For instance "validate extension properties" what scenario of the debranded OpenAI does it light up? My understanding is it is from Azure OpenAI SDK

Reply:
I stand corrected (and I learned me something... I've been doing this wrong.)